plutus-pab/demo/pab-nami/client/src/Cardano/Wallet/Nami.js

```diff
@@ -0,0 +1,115 @@
+"use strict";
+
+const buildTx = require('../../../lib/nami-wallet/src/api/extension/wallet.js').buildTx;
+const Loader = require('../../../lib/nami-wallet/src/api/loader.js').default;
+
+exports.enableImpl = () => window.cardano.enable()
+
+exports.isEnabled = window.cardano.isEnabled
+
+// TODO This should belong in the API of Nami wallet
+exports.getWalletIdImpl = () => {
+  return Promise.all([window.cardano.getUsedAddresses(), window.cardano.getUnusedAddresses()])
+    .then(([walletUsedAddresses, walletUnusedAddresses]) => {
+      const addresses = walletUnusedAddresses.concat(walletUsedAddresses)
+      return require('blake2b')(20)
+        .update(Buffer.from(addresses.map(a => a.to_bech32).join('')))
+        .digest('hex')
+    })
+}
```

Comment:
@koslambrou does this generate a WalletId compatible with the WBE?

Reply:
yes this uses the same hash algorithm as the one used in WBE